### PR TITLE
Audit and refactor framework settings

### PR DESCRIPTION
**Pull request by barrymcgee**
## Done

- Remove global settings that are too wide ranging in scope
- Split out remaining variables into distinct categories based on function
- Rename all variables to prefix them with their category
- Update `docs` with changes
- Update Release Notes to document changes

## QA

* Sanity check code changes
* Pull down code and run `gulp build`
* Ensure build passes

Fixes: #386

### CODE REVIEW

File: V1_RELEASE_NOTES.md [hunk 1]
```diff
@@ -3,3 +3,46 @@
...
+* `$success` is now `$color-success`
+* `$link-color` is now `$color-highlight`
+
+* `$gutter-width` is now `$grid-gutter-width`
+* `$columns` is now `$grid-columns`
+* `$site-max-width` is now `$grid-max-width`
+
+* `$navigation-threshold` has been replaced with `$breakpoint-small`
+
+* `$asset-server` is now `$assets-server`
```
Comment (anthonydillon) [inline V1_RELEASE_NOTES.md]: I would like to rename this one too `$assets-path` as it is used to link to local assets when the site news to work offline.

File: docs/settings/_assets-settings.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+collection: settings
+title: Assets settings
+---
+
+Setting  | Default Value
+ ------------- | -------------
+$assets-server:   | https://assets.ubuntu.com/v1/
```
Comment (anthonydillon) [inline docs/settings/_assets-settings.md]: See my note in the change log

File: docs/settings/_font-settings.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+---
+collection: settings
+title: Assets settings
+---
+
+Setting  | Default Value
+ ------------- | -------------
+$font-assets-path   | 'https://assets.ubuntu.com/sites/ubuntu/latest/u/'
```
Comment (anthonydillon) [inline docs/settings/_font-settings.md]: This is the old asset path. It should contain `v1`

File: docs/settings/_font-settings.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+$font-base-family   | 'Ubuntu, Arial, "libra sans", sans-serif'
+$font-base-size   | 1rem
```
Comment (anthonydillon) [inline docs/settings/_font-settings.md]: This will set the body font size to `1rem` we should set the body font size as a px value and use rem from there

**Comment by anthonydillon** [comment]
Thanks Barry, few comments.

**Comment by barrymcgee** [comment]
@anthonydillon I was wondering why we needed both, spoke to Graham and he explained the legacy issue. I'm now removed '$font-assets-path' entirely and updated all references to use '$assets-path'.

Ready for re-review.

**Comment by anthonydillon** [comment]
Perfect, thank you :+1:

**Pull request merged**